### ISSUE TITLE
Analyzer should make use of libraries.json file

### ISSUE THREAD

**Issue by stereotype441** [dart-lang/sdk#35081]
The common front end (and all the tools built from it) use a `libraries.json` file to locate SDK libraries (see documentation here: https://github.com/dart-lang/sdk/blob/master/pkg/front_end/lib/src/base/libraries_specification.dart#L5).  The analyzer still uses the old `libraries.dart` mechanism.  We should migrate the analyzer to use `libraries.json`.

This is the analyzer portion of #28836.

CC @sigmundch

**Comment by stereotype441** [comment]
Internal doc discussing details: https://docs.google.com/document/d/1unFHGfl4Xbw-U0Ass2OnV1gs9Ri2N-DA1GPJiLh7Des/edit

**Comment by srawlins** [comment]
@devoncarew was looking at this recently, and I think landed some preliminary work.

**Comment by devoncarew** [comment]
Yup, I have a CL outstanding which does a portion of the work to switch over: https://dart-review.googlesource.com/c/sdk/+/289401.
